src/ucd-data-fetch.c

```diff
@@ -420,11 +431,12 @@ int main(int argc, char *argv[]) {
 
 finish:
 
-	/* Don't run ucd for the test template */
-	if (strcmp(config[conf].name, "test") != 0) {
-		(void) execl(BINDIR "/ucd", BINDIR "/ucd", "-u", outpath, (char *)NULL);
-		FAIL("exec()");
+	snprintf(ucd, sizeof(ucd)-1, "%s/%s", BINDIR, "ucd");
+	if (realpath("/proc/self/exe", program)) {
+		snprintf(ucd, sizeof(ucd)-1, "%s/%s", dirname(program), "ucd");
 	}
```

Comment:
I'm not sure why this is needed - I do not think we should validate `ucd` when testing `ucd-data-fetch` here. In any case, we should never do this when `config[conf].name == "test"` because should protect against this daemon executing anything else than `/usr/bin/ucd` with sensitive data.

Better to just exit in case we've detected that we're running in `test` mode.